### PR TITLE
0.10.0 build 1 and examples package

### PR DESCRIPTION
**Pull request by Yoshanuikabundi**
The  `openff-toolkit` repository has a number of example notebooks that demonstrate its use, and we're about to put a bunch of effort into improving them and adding more. In openforcefield/openff-toolkit#888, I investigated adding tooling to simplify the use of the examples, and we discovered that they were not distributed with the toolkit by Conda. This PR adds a new subpackage, `openff-toolkit-examples`, that includes the examples themselves as well as depending on the packages required to run them. In the future, this package may also include a helper application to run the examples, but using this packaging method allows us to let Conda manage package management.

Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)

### CODE REVIEW

**Comment by Yoshanuikabundi** [comment]
@j-wags I got it working! With this PR, there is a `openff-toolkit-examples` package that installs the dependencies of the examples, and installs the examples themselves to `$CONDA_PREFIX/share/openff-toolkit/examples`.

My questions:
- Is that the best path to hold the examples?
- What tests should I add? Is just replicating all the notebook tests from upstream appropriate? I think the point is just to make sure all the dependencies are there and all the files are present but not sure what the best way to accomplish this is, since the aforementioned tests take about 10 minutes.

**Comment by j-wags** [comment]
To keep things in one place -- Discussion and next steps are in today's [meeting notes](https://openforcefield.atlassian.net/wiki/spaces/IN/pages/1652785167/2021-04-28+Mitchell+Wagner+Check-in)

**Comment by Yoshanuikabundi** [comment]
*\*Wipes sweat from brow*

Added the test checking that the installed file tree matches the source tree. It seems to be working!

File: recipe/meta.yaml [hunk 1]
```diff
@@ -8,10 +8,10 @@ package:
 source:
   - path: ./build_base.sh
   - url: https://github.com/openforcefield/openff-toolkit/archive/{{ version }}.tar.gz
```
Comment (j-wags) [inline recipe/meta.yaml:10]:
```suggestion
  - url: https://github.com/openforcefield/openff-toolkit/archive/refs/heads/examples_helper.zip
```

File: recipe/test_examples_copied.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    installed_tree = file_tree(installed_path)
+
+    print("Source repository examples tree:", *repo_tree, sep="\n    ")
+    print("Installed examples tree:", *installed_tree, sep="\n    ")
+
+    assert repo_tree, "Source repository tree is empty"
+    assert (
+        repo_tree == installed_tree
+    ), "Source repository tree doesn't match installed tree"
+
```
Comment (j-wags) [inline recipe/test_examples_copied.py:28]: add "run the examples copier and make sure everything gets copied out" test here

File: recipe/meta.yaml [hunk 3]
```diff
@@ -8,10 +8,10 @@ package:
...
-    sha256: 5bea41fdf5af5ffec8ca5203b714d0c7359507762df919cd42d1fec3b2de1686
+    sha256: b15aec2b0d4ae6384d96a6b53178a22be95ac39760b3a534ba740463eb819a27
```
Comment (j-wags) [inline recipe/meta.yaml:11]: Get this by running `curl -sL https://github.com/openforcefield/openff-toolkit/archive/refs/heads/examples_helper.zip | openssl sha256`

```suggestion
    sha256: 9e7b43a3e79760b3293c5ddb6556feb4196025e1f3325c3866cdc997c2d6b073
```

**Review by j-wags — COMMENTED** [review]
This should let us test from a branch of the toolkit -- We need to make sure to point this back to the release URL before we merge!!

**Comment by Yoshanuikabundi** [comment]
This is all updated and working with the latest version of the examples helper as of this comment. PR for the helper: https://github.com/openforcefield/openff-toolkit/pull/934

I did notice that the packages all still depend on `python >= 3.6`. Should we update this to `python >= 3.7` while we're here?

**Review by j-wags — COMMENTED** [review]
Looks great. Thanks for all the iterating on this @Yoshanuikabundi!

I think we need to merge this BEFORE the next release. As long as we revert the version+build, there shouldn't be any packages built or tests run. I'll give it a shot now :-)

**Comment by j-wags** [comment]
@conda-forge-admin, please rerender

**Comment by j-wags** [comment]
@conda-forge-admin, please rerender

**Pull request merged**